packages/fractal-page-object/src/-private/types.ts

```diff
@@ -9,3 +9,11 @@ export const CLONE_WITH_INDEX = Symbol('withIndex');
 export type PageObjectConstructor<T extends PageObject> = new (
   ...args: ConstructorParameters<typeof PageObject>
 ) => T | PageObject;
+
+/**
+ * A utility interface that can be unioned with a {@link PageObject}
+ * to denote that the `element` is _known_ to exist
+ */
+export interface WithElement {
```

Comment:
We already have a [WithElement](https://github.com/bendemboski/fractal-page-object/blob/b23244ea3ba195876abcbe691e166841b390ad46/packages/fractal-page-object/src/-private/array-stub.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L8) interface that's more targeted to `PageObject` and therefore IMO easier to use for its intended purpose. So let's update this one to look like that one, and then make `array-proxy.ts` import/use this instead of having its own.